src/v3/src/components/PIVButton/PIVButton.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box } from '@mui/material';
+import { Button, CircularLoadIndicator } from '@okta/odyssey-react';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+
+import { useWidgetContext } from '../../contexts';
+import {
+  ClickHandler,
+  PIVButtonElement,
+  UISchemaElementComponent,
+} from '../../types';
+import { getTranslation } from '../../util';
+
+const PIVButton: UISchemaElementComponent<{
+  uischema: PIVButtonElement
+}> = ({ uischema }) => {
+  const { translations = [] } = uischema;
+  const { options } = uischema;
+
+  const btnLabel = getTranslation(translations, 'label');
+
+  const { setMessage, loading } = useWidgetContext();
+  const [waiting, setWaiting] = useState<boolean>(false);
+
+  const showLoading = waiting || loading;
+
+  const initCertPrompt = () => {
+    setWaiting(true);
+    setMessage(undefined);
+
+    options.onClick();
+  };
```

Comment:
This is named `initCertPrompt` but none of the enclosed code is clearly doing that. I realize that `options.onClick()` will actually be what does that, and will probably always be the same function, but still feels a little strange to me. Any thoughts? I'm thinking someone looking through this in the future will be confused at first.

Reply:
Totally agree with this, this doesn't make much sense in this context, what I ended up doing was moving the cert trigger directly into this `PIVButton` component. The reason I did this was because this component is pretty specific to PIV, I contemplated maybe re-using it for other IDP buttons but I haven't researched those enough to understand the difference, I figured, when we get to that implementation, we can make the decision and refactor this component along with adding the `onClick` to the interface if needed. But since this is the only thing using it now, I thought it made sense to add it directly to the button that uses. 

What do you think?

Reply:
I think that's okay for now - it does feel like this component can become more generic as one that always performs an action on load and then can redo that action on click.